transformations/transformations/single_column_coalesced.py

```diff
@@ -728,9 +728,8 @@ def hoist_temporary_column_arrays(cls, routine, call, horizontal, vertical, bloc
         v_end = arg_map[kernel.variable_map[horizontal.bounds[1]]]
         bounds = sym.LoopRange((v_start, v_end))
         vector_loop = ir.Loop(variable=v_index, bounds=bounds, body=(new_call,))
-        call_map[call] = vector_loop
 
-        routine.body = Transformer(call_map).visit(routine.body)
```

Comment:
Why is this Transformer call being hoisted here? The new control flow is really counter-intuitive (`if call_map:` :raised_eyebrow: ), so I'm again not seeing how this is related to the removal of the `expr.source` property?